modules/tinymce/src/core/test/ts/browser/selection/MultiClickSelectionTest.ts

```diff
@@ -1,32 +1,128 @@
 import { describe, it } from '@ephox/bedrock-client';
 import { TinyAssertions, TinyHooks, TinySelections } from '@ephox/wrap-mcagar';
+import { assert } from 'chai';
 
+import DOMUtils from 'tinymce/core/api/dom/DOMUtils';
 import Editor from 'tinymce/core/api/Editor';
+import { findClosestBlockRange } from 'tinymce/core/selection/MultiClickSelection';
+
+import * as ViewBlock from '../../module/test/ViewBlock';
 
 describe('browser.tinymce.core.selection.MultiClickSelectionTest', () => {
-  const hook = TinyHooks.bddSetupLight<Editor>({
-    base_url: '/project/tinymce/js/tinymce'
-  }, [], true);
+  describe('findClosestBlockRange test', () => {
+    const DOM = DOMUtils.DOM;
+    const viewBlock = ViewBlock.bddSetup();
+    const getRoot = viewBlock.get;
+    const setupHtml = viewBlock.update;
+
+    const createRange = (sc: Node, so: number, ec: Node, eo: number) => {
+      const rng = DOM.createRng();
+      rng.setStart(sc, so);
+      rng.setEnd(ec, eo);
+      return rng;
+    };
+
+    const assertRange = (expected: Range, actual: Range) => {
+      assert.strictEqual(actual.startContainer, expected.startContainer, 'startContainers should be equal');
+      assert.strictEqual(actual.startOffset, expected.startOffset, 'startOffset should be equal');
+      assert.strictEqual(actual.endContainer, expected.endContainer, 'endContainer should be equal');
+      assert.strictEqual(actual.endOffset, expected.endOffset, 'endOffset should be equal');
+    };
+
+    const testFindClosestBlockRange = (startRng: Range, expectedRng: Range) =>
+      assertRange(expectedRng, findClosestBlockRange(startRng, getRoot()));
+
+    it(`should return the range with the whole text content of the block`, () => {
+      setupHtml('<p>aaa bIb ccc</p>');
+      const textNode = getRoot().firstChild.firstChild;
+      const startRng = createRange(textNode, 4, textNode, 7);
```

Comment:
I guess you could use Agar Cursors here or sugar Path directly to resolve paths to ranges. It would avoid having these long node.firstNode.firstNode.childNodes[1] chains since then you can have a path like [0, 0, 1] to get to the same place. So if you had a way to just take paths and offsets and produce dom ranges it would fit into the existing tests. Something like this `const expectedRng = resolveRange([ 0, 0 ], 0, [ 0, 0 ], 11)` to get a range of the whole range node but maybe my brain has just gotten more used to reading that syntax than having traversing node paths. I kind of like that this style is also explicitly named nodes as variables.

- [ ] Consider using path based assertion like other tests